src/geoops/transform.jl

```diff
@@ -0,0 +1,100 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    @transform(data, :newcol₁ = expr₁, :newcol₂ = expr₂, ..., :newcolₙ = exprₙ)
+
+Return a new data object with `data` columns and new columns `newcol₁`, `newcol₂`, ..., `newcolₙ`.
+The new columns are calculated using their respective expressions.
+In expressions the `data` columns are represented by `Symbols` and the operations use `broadcast` by default.
+
+# Examples
+
+```julia
+@transform(data, :z = :x + 2*:y)
+@transform(data, :w = :x^2 - :y^2)
+@transform(data, :sinx = sin(:x), :cosy = cos(:y))
+```
+"""
+macro transform(data::Symbol, exprs...)
+  splits = map(expr -> _split(data, expr), exprs)
+  colnames = first.(splits)
+  colexprs = last.(splits)
+  texpr = :(GeoStatsBase._transform($data, [$(colnames...)], [$(colexprs...)]))
+  esc(texpr)
+end
+
+function _transform(data::D, tnames, tcolumns) where {D<:Data}
+  dom   = domain(data)
+  table = values(data)
+
+  cols    = Tables.columns(table)
+  names   = Tables.columnnames(cols) |> collect
+  columns = [Tables.getcolumn(cols, nm) for nm in names]
+
+  @assert isdisjoint(tnames, names) "Invalid column names"
+
+  newnames   = [names; tnames]
+  newcolumns = [columns; tcolumns]
+
+  𝒯 = (; zip(newnames, newcolumns)...)
+  newtable = 𝒯 |> Tables.materializer(table)
+
+  newdata = Dict(paramdim(dom) => newtable)
```

Comment:
we only use the term "data" for geospatial data. Here we are generating a dictionary of tables. Try to follow the naming convention in other parts of the code.